src/App.tsx

```diff
@@ -1,61 +1,133 @@
 import './App.scss';
+import React, { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { Todo } from './types/Todos';
+import { TodoList } from './components/TodoList';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+function getUserById(id: number) {
+  return usersFromServer.find((user) => user.id === id) || null;
+}
+
+const todosWithUsers: Todo[] = todosFromServer.map((todo) => (
+  {
+    ...todo,
+    user: getUserById(todo.userId),
+  }
+));
+
+function getMaxId(todos: Todo[]): number {
+  return Math.max(...todos.map((todo) => todo.id)) + 1;
+}
 
 export const App = () => {
+  const [todos, setTodos] = useState(todosWithUsers);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasUserError, setHasUserError] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+(event.target.value));
+    setHasUserError(false);
+  };
+
+  const clearAll = () => {
+    setTitle('');
+    setUserId(0);
+    setHasTitleError(false);
+    setHasUserError(false);
+  };
+
+  const handleFormSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (userId && title.trim()) {
```

Comment:
```suggestion
    if (!userId || !title) {
      setHasUserError(!userId);
      setHasTitleError(!title);
      
      return;
    }
```

Don't do unnecessary actions